### ISSUE TITLE
Add Semantic modifiers for keyword arguments in function or class call.

### ISSUE THREAD

**Issue by mvanderlee** [microsoft/pylance-release#4944]
This may be related to #2913

# Goal
Allow keyword arguments to have a unique color to make something like this more readable

![image](https://github.com/microsoft/pylance-release/assets/918128/58462d28-7f69-4e5d-af00-c7dee9b87871)

An added modifier like `function-call` so we can color `parameter.function-call` should do the trick.

# Observations

Classes and Dataclasses kwargs have different semantic token types
A class gets token type `parameter` and a dataclass gets `property`. An arg is of type `parameter` in both cases. I suspect this is because technically a dataclass does have properties, and the rest is autogenerated. However, in the context of instantiating the instance, it is a parameter to the function call and should get the `parameter` token.

![image](https://github.com/microsoft/pylance-release/assets/918128/f0ffac31-6822-4d03-bdaa-001ab3c33def)
![image](https://github.com/microsoft/pylance-release/assets/918128/792fc88b-42a4-469c-8e3c-3dd09d89cdb9)

## Example code
```python
from dataclasses import dataclass

class RegularClass:
    def __init__(self, foo: str, bar: str = None):
        self.foo = foo
        self.bar = bar

@dataclass
class Dataclass:
    foo: str
    bar: str = None
          
def my_function(foo_arg: str, bar_kwarg: str = None):
    rc = RegularClass(
        foo_arg,
        bar=bar_kwarg,
    )

    dc = Dataclass(
        foo_arg,
        bar=bar_kwarg,
    )
```

**Comment by rchiodo** [comment]
Thanks for the issue, but I'm confused on what the ask is.

In this code here, you want it to say parameter instead of property?

![image](https://github.com/microsoft/pylance-release/assets/19672699/089d6fed-be1e-4b63-a8a1-09d0c872e1f1)

Or is that in addition to adding something extra to your first example?

In your first example it's unclear to me which parameters would have `function-call` appended to their tokens.

**Comment by mvanderlee** [comment]
The ask is, I want to be able to achieve this effect using semantic colorization. (this theme uses textmate)
![image](https://github.com/microsoft/pylance-release/assets/918128/d659c994-5fe7-46b0-b006-38881e6d7356)

1. All function call parameters should have a `function-call` modifier. I think a class instantiation should be handled like a function call and get the same modifier.
2. I think that `bar` should be a parameter instead of a property. Because in this context, it's a function parameter.

PS: I don't care what the modifier is called, but textmate scopes already have `function-call` so reusing it makes sense to me?

**Comment by rchiodo** [comment]
Thanks, that makes more sense.

The function-call thing seems reasonable. 

Not sure about changing the properties to parameters. They'd only be parameters in this case I think?

```python
class MyClass():
   x: int
   y: int
   def __init__(self, x, y):
        self.x = x
        self.y = y

a = MyClass(x=4, y=5)
```

Which it looks like already works.

**Comment by rchiodo** [comment]
And if we added the function-call special token, those would end up as `property.function-call`?

**Comment by mvanderlee** [comment]
Right, but a dataclass autogenerates the `__init__` function. Pylance is just not picking up on it. 

I strongly believe that whenever I call a function or class, we are only talking about parameters. Never properties. 
A property would be whenever I use dot notation.

```
from dataclasses import dataclass

@dataclass
class Dataclass:
    foo: str
    bar: str = None
          
dc = Dataclass(
    foo_arg,
    bar=bar_kwarg,   # parameter
)

dc.bar = 'foo'   # property
```

Dataclasses could get more advanced when we talk about ClassVars, and `field(init=False)`. But I'm not sure we care for this purpose? Essentially any `key=value` inside brackets would be a parameter with value. Not a property.

**Comment by heejaechang** [comment]
Repro workspace for investigation: https://github.com/heejaechang/pylance-repro/tree/main/pylance-issues/4944

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2026.2.108, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20262108-9-July-2026-PreRelease)
